jib-core/src/main/java/com/google/cloud/tools/jib/image/ReproducibleLayerBuilder.java

```diff
@@ -34,6 +38,36 @@
  */
 public class ReproducibleLayerBuilder {
 
+  // Uses the current directory to act as the file input to TarArchiveEntry (since all directories
+  // are treated the same in TarArchiveEntry).
+  private static final File DIRECTORY_FILE = Paths.get(".").toFile();
+
+  /** Decorates an object with a separate object for {@link #hashCode}. */
```

Comment:
I'd appreciate detail as to why this class is required.  Is `Path#hashCode()` possibly inconsistent?  (`UnixPath#hashCode()` looks fine)